cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1170,8 +1170,7 @@ private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> onReady(Reconcilia
                             || kafka.getStatus().getConditions() == null
                             || kafka.getStatus().getConditions().stream().noneMatch(condition -> condition.getType().equals("Ready") && condition.getStatus().equals("True"))) {
                         LOGGER.warnCr(reconciliation, "Kafka cluster is not Ready");
-                        KafkaRebalanceStatus status = new KafkaRebalanceStatus();
-                        return updateStatus(reconciliation, kafkaRebalance, status,
+                        return updateStatus(reconciliation, kafkaRebalance, null,
```

Comment:
Here you are skipping the status update regardless of the current rebalance state, which you need to take into account to only skip the status update when the current state is ready.